### ISSUE TITLE
avm2: Panic when loader dispatch event to another eventdispatcher

### ISSUE THREAD

**Issue by AllinolCP** [ruffle-rs/ruffle#10748]
### Describe the bug

when using a loader and passing the event into another eventdispatcher panics ruffle
[here is a minimal reproduction of the issue](https://github.com/ruffle-rs/ruffle/files/11279801/main.zip)
here is the panic

```
Error: panicked at 'No caller domain available - use Activation::from_domain when constructing your domain', core/src/avm2/activation.rs:676:28
    at O.n.wbg.__wbg_new_15d3966e9981a196 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.dcbb4a112d0621488d8f.js:1:47949)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::hbac0af9b7f44ed2e (wasm://wasm/02a77236:wasm-function[8135]:0x7ee656)
    at std::panicking::rust_panic_with_hook::hc53aea0352e77326 (wasm://wasm/02a77236:wasm-function[5463]:0x7272f5)
    at core::panicking::panic_fmt::hf4a9df75710ece83 (wasm://wasm/02a77236:wasm-function[8017]:0x7e92f3)
    at core::option::expect_failed::h8f07431afd4c5625 (wasm://wasm/02a77236:wasm-function[9308]:0x815ae9)
    at ruffle_core::avm2::function::Executable::exec::h459ce63d33fcd2f3 (wasm://wasm/02a77236:wasm-function[921]:0x2f9159)
    at <ruffle_core::avm2::object::Object as ruffle_core::avm2::object::TObject>::call::h4aec07be6c66dfa0 (wasm://wasm/02a77236:wasm-function[945]:0x3091ec)
    at ruffle_core::avm2::events::dispatch_event_to_target::hc9fd98c358ca4352 (wasm://wasm/02a77236:wasm-function[886]:0x2e3abd)
    at ruffle_core::avm2::events::dispatch_event::h48ad47345cdb0200 (wasm://wasm/02a77236:wasm-function[1040]:0x33e7e5)
    at ruffle_core::avm2::Avm2::dispatch_event::hd8c3385336e5af8e (wasm://wasm/02a77236:wasm-function[1248]:0x3a0ad7)
```

### Expected behavior

the event should dispatch instead of panicking

### Affected platform

Desktop app

### Operating system

Windows 11

**Comment by AllinolCP** [comment]
problematic line has been traced down to this [line](https://github.com/ruffle-rs/ruffle/blob/4f4327b152087379bae0d566a5d80aa7c9b9c477/core/src/avm2/function.rs#L116)

**Comment by Lord-McSweeney** [comment]
Actually the relevant line is here: https://github.com/ruffle-rs/ruffle/blob/4f4327b152087379bae0d566a5d80aa7c9b9c477/core/src/avm2.rs#L317

It should be constructing `Activation::from_domain`, though I'm not sure whether that's the script domain or the global domain.

**Comment by AllinolCP** [comment]
> Actually the relevant line is here: https://github.com/ruffle-rs/ruffle/blob/4f4327b152087379bae0d566a5d80aa7c9b9c477/core/src/avm2.rs#L317
> 
> It should be constructing `Activation::from_domain`, though I'm not sure whether the domain should be the script domain or the global domain.

this is odd, i hack-fix avm2/function.rs:117 and it worked

**Comment by Lord-McSweeney** [comment]
Fixed by #10811.
